### PR TITLE
Remove window._brave_solana

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/25645

1. We now use solana-web3.js minified js exported solanaWeb3 to generate `PublicKey` and `Transaction`.
2. Script resources is packed in `IDR_BRAVE_WALLET_SOLANA_WEB3_JS`
3. `//brave/components/brave_wallet/resources/solana_provider_internal.js` is completely removed so iOS needs to load the script through resource in a similar way as we load it in v8 and access `solanaWeb3.PublicKey` and `solanaWeb3.Transaction` directly.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Setup wallet 
2. Open any sites and open dev console
3. Type `window._brave_solana` should get `undefined`
4. Navigate to https://codesandbox.io/s/github/darkdh/solana-provider-test
5. Should be able to `Connect`
6. After connected, open dev console and type` window._brave_solana` should still get `undefined`
7. Test `SignTransaction` and `SignAllTransactions` should be successful

### CODE REVIEW

**Review by bbondy — APPROVED** [review]

File: components/brave_wallet/renderer/js_solana_provider.cc [hunk 1]
```diff
@@ -16,16 +16,20 @@
 #include "brave/components/brave_wallet/renderer/resource_helper.h"
 #include "brave/components/brave_wallet/renderer/v8_helper.h"
 #include "brave/components/brave_wallet/resources/grit/brave_wallet_script_generated.h"
+#include "components/grit/brave_components_resources.h"
```
Comment (bbondy) [inline components/brave_wallet/renderer/js_solana_provider.cc:19]: Need to add the deps for this per CI

File: ios/browser/api/brave_wallet/brave_wallet_api.mm [hunk 2]
```diff
@@ -26,8 +27,8 @@
     @"ethereum_provider.js";
 BraveWalletProviderScriptKey const BraveWalletProviderScriptKeySolana =
     @"solana_provider.js";
-BraveWalletProviderScriptKey const BraveWalletProviderScriptKeySolanaInternal =
-    @"solana_provider_internal.js";
+BraveWalletProviderScriptKey const BraveWalletProviderScriptKeySolanaWeb3JS =
```
Comment (StephenHeaps) [inline ios/browser/api/brave_wallet/brave_wallet_api.mm:30]: The `ios/browser/api/brave_wallet/brave_wallet_api.h` header file will need updated for this name change too:
https://github.com/brave/brave-core/blob/68e162730a2fda99cf61ff107bf5ae44a5708117/ios/browser/api/brave_wallet/brave_wallet_api.h#L25-L26

**Review by StephenHeaps — CHANGES_REQUESTED** [review]

File: ios/browser/api/brave_wallet/brave_wallet_api.mm (hunk 2 above)
Comment (darkdh) [reply ios/browser/api/brave_wallet/brave_wallet_api.mm:30]: fixed in a90423f98bf26d062849f2ad624b9ba2cacc7863

Comment (kylehickinson) [reply ios/browser/api/brave_wallet/brave_wallet_api.mm:30]: Not fixed, ICYMI: https://github.com/brave/brave-core/commit/a90423f98bf26d062849f2ad624b9ba2cacc7863#r85274366

Comment (darkdh) [reply ios/browser/api/brave_wallet/brave_wallet_api.mm]: 24fcb4ffd34bfb42a6c45f4bc6f6b78e810789cc
should fix it now

File: ios/browser/api/brave_wallet/brave_wallet_api.mm [hunk 3]
```diff
@@ -158,13 +159,11 @@ - (NSString*)resourceForID:(int)resource_id {
...
-            std::make_pair(
-                BraveWalletProviderScriptKeySolana,
-                IDR_BRAVE_WALLET_SCRIPT_SOLANA_PROVIDER_SCRIPT_BUNDLE_JS),
-            std::make_pair(
-                BraveWalletProviderScriptKeySolanaInternal,
-                IDR_BRAVE_WALLET_SCRIPT_SOLANA_PROVIDER_INTERNAL_SCRIPT_BUNDLE_JS)};
+        return {std::make_pair(
+                    BraveWalletProviderScriptKeySolana,
+                    IDR_BRAVE_WALLET_SCRIPT_SOLANA_PROVIDER_SCRIPT_BUNDLE_JS),
+                std::make_pair(BraveWalletProviderScriptKeySolanaWeb3JS,
```
[the remark this answers is not in the document]
Comment (darkdh) [reply ios/browser/api/brave_wallet/brave_wallet_api.mm]: https://github.com/brave/brave-core/pull/15247/commits/e336adddfd079fc781690934ad434acf6520c2c8
lol

**Review by StephenHeaps — APPROVED** [review]

**Review by thypon — APPROVED** [review]
Tested dynamically, GTG!

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.46.40 Chromium: 106.0.5249.103 (Official Build) nightly (64-bit)
-- | --
Revision | `182570408a1f25ab2731ef5f283b918df9b9f956-refs/branch-heads/5249_91@{#6}`
OS | Windows 11 Version 22H2 (Build 22623.730)

- Verified steps from PR
- Verified able to connect to Dapp 
- Verified able to submit `Sign Transaction` & `Sign All Transaction`

https://user-images.githubusercontent.com/17010094/194483611-d1e3a11a-cbb7-416a-9cfd-51820e5e4dd2.mp4

**Pull request merged**